ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -27,79 +29,150 @@ public class NotificationImportService : INotificationImportService
+        public async Task<IList<ImportResult>> ImportByDateAsync(PerformContext context, string requestId, DateTime rangeStartDate, DateTime rangeEndDate)
+        {
+            _logger.LogInformation(context, requestId, $"Request to import by Date started");
+            _logger.LogInformation(context, requestId, $"Importing notifications in range from {rangeStartDate.Date} to {rangeEndDate.Date}");
+
+            var notificationsGroups = await _notificationMapper.GetNotificationsGroupedByPatient(rangeStartDate, rangeEndDate);
+
+            // Filter out notifications that already exist in ntbs database
+            var notificationsGroupsToImport = new List<List<Notification>>();
+            foreach (var notificationsGroup in notificationsGroups)
+            {	            
+                var legacyIds = notificationsGroup.Select(x => x.LegacyId).ToList();
+                var notificationsToImport = GetNotificationsToImport(context, requestId, legacyIds);
+                if (notificationsToImport.Count() == notificationsGroup.Count())
+                {
+                    notificationsGroupsToImport.Add(notificationsGroup);
+                }
+                else if (notificationsToImport.Count() != 0)
+                {
+                    _logger.LogFailure(context, requestId, $"Invalid state. Some notifications already exist database. Manual intervention needed");
+                }
+            }
+            
+            var importResults = new List<ImportResult>();
+            if (notificationsGroupsToImport.Count() > 0)
             {
-                _logger.LogInformation(requestId, $"Notification with Id={notificationId} already exists in database");
-                return null;
+                // Validate and Import valid notifications
+                importResults = notificationsGroupsToImport
+                    .Select(notificationsGroup => ValidateAndImportNotificationGroupAsync(context, requestId, notificationsGroup))
+                    .Select(t => t.Result)
+                    .ToList();
             }
 
-            var notifications = (await _notificationMapper.Get(notificationId)).ToList();
-            if (notifications.Count == 0)
+            _logger.LogInformation(context, requestId, $"Request to import by Date finished");
+            return importResults;
+        }
+        
+        public async Task<IList<ImportResult>> ImportByLegacyIdsAsync(PerformContext context, string requestId, List<string> notificationIds)
+        {
+            _logger.LogInformation(context, requestId, $"Request to import by Id started");
+
+            // Filtering out notifications that already exist in ntbs database.
+            var notificationIdsToImport = GetNotificationsToImport(context, requestId, notificationIds);
+
+            var importResults = new List<ImportResult>();
+            if (notificationIdsToImport.Count() > 0)

```

Comment:
We used to log in the `else` case that ` $"No notifications found with Id={notificationId}"` - might be worth keeping that around?